pkg/api/httpcontext/httpcontext.go

```diff
@@ -0,0 +1,76 @@
+package httpcontext
+
+import (
+	"math/rand"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+const header = "X-Trace"
+
+var (
+	lock     sync.RWMutex
+	requests map[*http.Request]api.Context
```

Comment:
Suggestion: make a type having these as members; that will be easier to test. Functions that access these should have the new type as a receiver.  You can make a default instance of it if you wish.